### PR TITLE
Add support for combo counter from legacy mania skins

### PR DESCRIPTION
**Pull request by TheEZIC**
Move all code from LegacyComboCounter to LegacyBaseComboCounter. We don't need ISerialisableDrawable implementation for LegacyManiaComboCounter. I made it like this because we can't move mania ComboCounter without configuration changes.

Made LegacyManiaComboCounter from LegacyBaseComboCounter.

Move logic of hiding default combo counter element from CatchLegacySkinTransformer to LegacySkinTransformer because we need to hide default combo counter in mania as well.

Add ShouldRoll property to BaseComboCounter. That property toggles rolling of combo counter and removes 'x' symbol at the end of it.

### CODE REVIEW

**Comment by longnguyen2004** [comment]
Is this related to #26254?

**Comment by bdach** [comment]
> Is this related to #26254?

It sure looks like this to me, and some of the changes made in this pull on a 5-second skim make it dead-on-arrival as far as I'm concerned.

The other series has the right idea on how to execute this. This PR does not.

File: osu.Game/Skinning/LegacyBaseComboCounter.cs
Comment (bdach) [inline osu.Game/Skinning/LegacyBaseComboCounter.cs:1]: This class should not exist

File: osu.Game/Skinning/LegacySkinTransformer.cs [hunk 1]
```diff
@@ -34,5 +37,29 @@ protected LegacySkinTransformer(ISkin skin)
 
             return base.GetSample(sampleInfo);
         }
+
+        protected Drawable? ReplaceBaseComboCounter(ISkinComponentLookup lookup)
```
Comment (bdach) [inline osu.Game/Skinning/LegacySkinTransformer.cs:41]: This method cannot exist

**Review by bdach — COMMENTED** [review]
Just to elaborate on what I find objectionable about this change for future reference.

We added ruleset-specific HUD layers for the express purpose of implementing this sort of thing. So bypassing them entirely via means such as the ones introduced in this PR will be met with instant rejection.

**Pull request closed without merging**